### PR TITLE
docs: cursor rules for tool generation from docs

### PR DESCRIPTION
**Pull request by KarthikBoddeda**
## Type of Change

- [x] Documentation update

## Testing

- [x] Manual testing

## Checklist

- [x] I have followed the code style of this project
- [x] I have added comments to code where necessary, particularly in hard-to-understand areas
- [x] I have updated the documentation where necessary
- [x] I have verified that my changes do not introduce new warnings or errors
- [x] I have checked for and resolved any merge conflicts

### CODE REVIEW

File: .cursor/rules/new-tool-from-docs.mdc [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+## Implementation Checklist
+
+This checklist **MUST** be included in the final response to verify all implementation steps are complete.
+IMPORTANT: Include this unchecked checklist at the END of the implementation, NOT at the beginning.
+
+- [ ] Implement tool function based on API docs
+- [ ] Register tool in tools.go
+- [ ] Create unit tests with full coverage (positive case, all negative cases, edge cases)
+- [ ] Update README.md and make sure the new additions are correctly formatted.
```
Comment (alok87) [inline .cursor/rules/new-tool-from-docs.mdc:38]: which README?

File: .cursor/rules/new-tool-from-docs.mdc [hunk 2]
```diff
@@ -0,0 +1,245 @@
+---
```
[the remark this answers is not in the document]
Comment (KarthikBoddeda) [reply .cursor/rules/new-tool-from-docs.mdc:1]: yeah, most of the instructions are specific to new rules. Will make it more generic or add a new prompt to edit/fix existing rules once we run into that use case

**Review by alok87 — APPROVED** [review]

File: .cursor/rules/new-tool-from-docs.mdc (hunk 1 above)
Comment (KarthikBoddeda) [reply .cursor/rules/new-tool-from-docs.mdc:38]: we have this section in the README to list out the available tools: https://github.com/razorpay/razorpay-mcp-server/blob/bafb8fba9619d872914e02ffd2ca5407b52a98ff/README.md#L5-L15

Will explicitly mention that.

**Review by alok87 — APPROVED** [review]
